src/test/java/org/jabref/logic/pdf/search/indexing/DocumentReaderTest.java

```diff
@@ -0,0 +1,102 @@
+
+        new DocumentReader(entry).readPdfContents();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void unknownFileTestShouldThrowIllegalArgumentException() throws IOException {
+        BibEntry entry = mock(BibEntry.class);
+        when(entry.getField(FieldName.FILE)).thenReturn(Optional.empty());
+
+        new DocumentReader(entry);
+    }
+
+    @Test
+    public void exampleTest() throws IOException {
+        Path example = Paths.get("src/test/resources/pdfs/example.pdf");
+
+        BibEntry entry = mock(BibEntry.class);
+        when(entry.getField(FieldName.FILE)).thenReturn(Optional.of(example.toString()));
+        when(entry.getCiteKeyOptional()).thenReturn(Optional.of("Example2017"));
+
+        Document doc = new DocumentReader(entry).readPdfContents();
+
+        assertEquals("Example2017", doc.get(SearchFieldConstants.KEY));
+        assertFalse(doc.get(SearchFieldConstants.CONTENT).isEmpty());
+        assertEquals("LaTeX with hyperref package", doc.get(SearchFieldConstants.CREATOR));
+    }
+
+    @Test
+    public void thesisExampleTest() throws IOException {
+        Path example = Paths.get("src/test/resources/pdfs/thesis-example.pdf");
+
+        BibEntry entry = mock(BibEntry.class);
+        when(entry.getField(FieldName.FILE)).thenReturn(Optional.of(example.toString()));
+        when(entry.getCiteKeyOptional()).thenReturn(Optional.of("ThesisExample2017"));
+
+        Document doc = new DocumentReader(entry).readPdfContents();
+
+        assertEquals("ThesisExample2017", doc.get(SearchFieldConstants.KEY));
+        assertFalse(doc.get(SearchFieldConstants.CONTENT).isEmpty());
+        assertEquals("LaTeX, hyperref, KOMA-Script", doc.get(SearchFieldConstants.CREATOR));
+    }
+
+    @Test
+    public void minimalTest() throws IOException {
+        Path example = Paths.get("src/test/resources/pdfs/minimal.pdf");
+
+        BibEntry entry = mock(BibEntry.class);
+        when(entry.getField(FieldName.FILE)).thenReturn(Optional.of(example.toString()));
+        when(entry.getCiteKeyOptional()).thenReturn(Optional.of("Minimal2017"));
+
+        Document doc = new DocumentReader(entry).readPdfContents();
+
+        assertEquals("Minimal2017", doc.get(SearchFieldConstants.KEY));
+        assertEquals("Hello World\n1\n", doc.get(SearchFieldConstants.CONTENT));
+        assertEquals("TeX", doc.get(SearchFieldConstants.CREATOR));
+    }
+
+    @Test
+    public void metaDataTest() throws IOException {
+        Path example = Paths.get("src/test/resources/pdfs/metaData.pdf");
+
+        BibEntry entry = mock(BibEntry.class);
+        when(entry.getField(FieldName.FILE)).thenReturn(Optional.of(example.toString()));
+        when(entry.getCiteKeyOptional()).thenReturn(Optional.of("MetaData2017"));
+
+        Document doc = new DocumentReader(entry).readPdfContents();
+
+        assertEquals("MetaData2017", doc.get(SearchFieldConstants.KEY));
```

Comment:
Wouldn't it make sense to compare two Document-objects? (Needs an equals method then)
e.g. Document expected = new Document(....)